ERCS/erc-6900.md

```diff
@@ -622,11 +622,21 @@ See `https://github.com/erc6900/reference-implementation`
+The standard introduces a concept of the execution manifest, which describes the execution functions, interface IDs, and hooks that should be installed on the account from an execution manifest. As part of constructing parameters to `installExecution()`, the SDK creates an `executionManifest` parameter that will specify the actions of the execution module being installed. SDKs will process the provided `executionManifest()` function of the underlying module, but the finished parameter will not necessarily be identical to the module’s return value of `executionManifest()`. Furthermore, this parameter is only going through very limited checks on-chain as part of the installation process. Therefore, the `executionManifest` parameter needs to be carefully constructed and verified before being used for installations. Furthermore, the `executionManifest` parameter of uninstallation should ideally be the same used in installation to not leave residual data.
+
+The standard supports a `isGlobalValidation` flag for validation functions, which means that this function is added to a global validation pool and can validate any of the execution functions that expose themselves to global validation via `allowGlobalValidation` flag. By default all native execution functions are allowed to be globally validated. Therefore, the wallets should be careful on what validation functions they allow to be installed that have global validation allowed, as that would allow these functions to validate all native functions by default and hence bypass any restrictions that have been added to protect these native/execution functions. In a sense these global validation functions gain root access to the account.
+
+### Module Developers
+
+The standard does not enforce any rules surrounding what data needs to be installed or uninstalled when a module is added/deleted from the account. Hence the `onUninstall()` function in various modules may leave behind residual state data, especially since the external call is only performed if `onInstallData` is specified. Furthermore, execution hooks linked to an uninstalled function may remain configured. This could pose security risks or lead to unexpected behavior in the case where the module is reinstalled with the same `entityId`. The danger is that previously set permissions or data may be unintentionally reused. Hence it is a good idea to fully uninstall all data linked to the smart module account including execution hooks.
```

Comment:
>Hence the `onUninstall()` function in various modules may leave behind residual state data, especially since the external call is only performed if `onInstallData` is specified.

I think we should rewrite this to:

"Hence the `onUninstall()` function in various modules may leave behind residual state data, especially since the external call may not be performed all."

@nikita-quantstamp this is just because the spec only states that accounts SHOULD call this function if the user requested it, but doesn't dive deep into what that entails.

Reply:
This makes sense, will update the wording